sqlglot/optimizer/qualify_columns.py

```diff
@@ -249,6 +250,12 @@ def _qualify_columns(scope, resolver):
                 raise OptimizeError(f"Unknown column: {column_name}")
 
         if not column_table:
+            if scope.pivots and not column.find_ancestor(exp.Pivot):
+                # If the column is under the Pivot expression, we need to qualify it
+                # using the name of the pivoted source instead of the pivot's alias
+                column.set("table", exp.to_identifier(scope.pivots[0].alias))
+                continue
```

Comment:
Take for example:

```sql
SELECT x, y FROM u PIVOT (SUM(f) FOR h IN ('x', 'y'));
```

Here, we want to qualify `f` and `h` using `u` instead of the pivot's (generated) alias `_q_0`. The call to `find_ancestor` solves this problem, because otherwise we'd get the (invalid) form:

```sql
PIVOT (SUM(_q_0.f) FOR _q_0.h IN ('x', 'y'))`
```